### PR TITLE
Fix x-scope not rendered in Resource UI. Fix NPE when updating legacy scopes.

### PR DESCRIPTION
**Pull request by dushaniw**
fix https://github.com/wso2/product-apim/issues/9002
fix https://github.com/wso2/product-apim/issues/9018

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java [hunk 1]
```diff
@@ -1274,12 +1274,16 @@ private boolean isDefaultGiven(String swaggerContent) throws APIManagementExcept
      */
     @Override
     public String processOtherSchemeScopes(String swaggerContent) throws APIManagementException {
-        if (!isDefaultGiven(swaggerContent)) {
-            Swagger swagger = getSwagger(swaggerContent);
-            swagger = processLegacyScopes(swagger);
+        Swagger swagger = getSwagger(swaggerContent);
+        Set<Scope> legacyScopes = getScopesFromExtensions(swagger);
+
+        if (!isDefaultGiven(swaggerContent) && legacyScopes.isEmpty()) {
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java:1280]: does legacyScopes will not null

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java [hunk 2]
```diff
@@ -1274,12 +1274,16 @@ private boolean isDefaultGiven(String swaggerContent) throws APIManagementExcept
...
             swagger = injectOtherScopesToDefaultScheme(swagger);
             swagger = injectOtherResourceScopesToDefaultScheme(swagger);
             return getSwaggerJsonString(swagger);
+        } else if (!legacyScopes.isEmpty()) {
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java:1284]: same goes here

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java (hunk 1 above)
Comment (dushaniw) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java:1280]: No. it is empty LinkedHashSet returning from getScopesFromExtensions

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java (hunk 2 above)
Comment (dushaniw) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java:1284]: No. it is empty LinkedHashSet returning from getScopesFromExtensions

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java (hunk 1 above)
Comment (dushaniw) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java:1280]: https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java#L421

Comment (tharindu1st) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java:1280]: ok

**Pull request merged**